diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index 5898b8c54b..2e7b49dd5e 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -2405,8 +2405,9 @@ if (is_win) {
         # Specifically tell the linker to perform optimizations.
         # See http://lwn.net/Articles/192624/.
         # -O2 enables string tail merge optimization in lld.
-        "-Wl,-O2",
+        "-Wl,-O2","-Wl,-mllvm,-march=znver3", "-Wl,-mllvm,-fp-contract=fast", "-Wl,-zcommon-page-size=2097152", "-Wl,-zmax-page-size=2097152",
       ]
+      common_optimize_on_cflags += ["-march=znver3", "-ffp-contract=fast",] # "-O3",
     }

     common_optimize_on_ldflags += [ "-Wl,--gc-sections" ]
